File: src/main/java/com/clearcont/clearcontapp/helpers/MonthAndCompany.java
package com.clearcont.clearcontapp.helpers;

import com.clearcont.clearcontapp.model.Empresa;
import com.clearcont.clearcontapp.repository.EmpresaRepository;
import com.vaadin.flow.component.UI;
import com.vaadin.flow.component.page.Page;
import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Component;

import java.util.function.Consumer;

@Component
public interface MonthAndCompany {
    String getMonth();
    
    Empresa getEmpresa();
    
    void setEmpresa(Empresa empresa);
    
    void setMonth(String month);
    
    default void getCompany(@NotNull EmpresaRepository empresaRepository, @NotNull Consumer<Empresa> callback) {
        UI ui = UI.getCurrent();
        Page page = ui.getPage();
        page.executeJs("return sessionStorage.getItem($0)", "company-name")
                .then(item -> {
                    setEmpresa(empresaRepository.findEmpresaByNomeEmpresa(item.asString()).orElse(new Empresa()));
                    callback.accept(getEmpresa());
                });
    }
    
    default void getMonth(@NotNull Consumer<String> callback) {
        UI ui = UI.getCurrent();
        Page page = ui.getPage();
        page.executeJs("return sessionStorage.getItem($0)", "month")
                .then(item -> {
                    setMonth(item.asString());
                    callback.accept(getMonth());
                });
    }
}

File: src/main/kotlin/br/com/clearcont/clearcontwebapp/helpers/MonthAndCompany.kt
package br.com.clearcont.clearcontwebapp.helpers

import com.clearcont.clearcontapp.model.Empresa
import com.clearcont.clearcontapp.repository.EmpresaRepository
import com.vaadin.flow.component.UI
import elemental.json.JsonValue
import org.springframework.stereotype.Component
import java.util.function.Consumer

@Component
interface MonthAndCompany {
    var month: String?

    var empresa: Empresa?

    fun getCompany(empresaRepository: EmpresaRepository, callback: Consumer<Empresa?>) {
        val ui = UI.getCurrent()
        val page = ui.page
        page.executeJs("return sessionStorage.getItem($0)", "company-name")
            .then { item: JsonValue ->
                empresa = empresaRepository.findEmpresaByNomeEmpresa(item.asString()).orElse(Empresa())
                callback.accept(empresa)
            }
    }

    fun getMonth(callback: Consumer<String?>) {
        val ui = UI.getCurrent()
        val page = ui.page
        page.executeJs("return sessionStorage.getItem($0)", "month")
            .then { item: JsonValue ->
                month = item.asString()
                callback.accept(month)
            }
    }
}
